File: src/1343_Number_of_Sub-arrays_of_Size_K_and_Average_Greater_than_or_Equal_to_Threshold/solution.py
## https://leetcode.com/problems/number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/

from typing import List


## TC: O(N)
## SC: O(1)
def numOfSubarrays(self, arr: List[int], k: int, threshold: int) -> int:
    n = len(arr)
    l = r = 0
    summary = arr[r]
    count = 0
    while r < n:
        if r - l + 1 < k:
            r += 1
            summary += arr[r]
        else:
            if summary / k >= threshold:
                count += 1
            summary -= arr[l]
            l += 1
            r += 1
            if r <= n - 1:
                summary += arr[r]
    return count

File: src/1343_Number_of_Sub-arrays_of_Size_K_and_Average_Greater_than_or_Equal_to_Threshold/solution.ts
// https://leetcode.com/problems/number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/

// TC: O(N)
// SC: O(1)

function numOfSubarrays(arr: number[], k: number, threshold: number): number {
  const n = arr.length;
  const target = k * threshold;
  let l = 0;
  let summary = 0;
  let count = 0;

  for (let r = 0; r < n; r++) {
    summary += arr[r];
    if (r - l + 1 < k) continue;
    if (summary >= target) count += 1;
    summary -= arr[l];
    l += 1;
  }

  return count;
}
